bazel/BUILD

```diff
@@ -7,6 +7,30 @@ exports_files([
     "sh_test_wrapper.sh",
 ])
 
+genrule(
+    name = "gnu_build_id",
+    outs = ["gnu_build_id.ldscript"],
+    cmd = """
+      echo --build-id=0x$$(
+          grep STABLE_BUILD_SCM_REVISION bazel-out/stable-status.txt \\
```

Comment:
How does this work when Envoy is used as a dependency in another project to assemble a binary? Are there any issues with relative paths? Can you validate with envoy-filter-example?

Reply:
I don't fully understand how the build-id feature is being used now, so I don't know what the desired behavior is when building from a higher-level repo.

Do you want the build ID to be Envoy's commit ID, or the commit ID of the root workspace? What if it's not being built from a Git repo -- should we make the build ID empty, or leave the flag unset, or ...?

Reply:
@mattklein123 or other Lyfters will have to answer this one, as they have a similar setup to envoy-filter-example and use this for production build stamping.

Reply:
I think right now it's the SHA of the envoy repo. Optimally, it would be the SHA of the owning repo, but whatever is fine with us as long as we have something.

Reply:
I've tweaked this a bit so it'll always use `BUILD_SCM_REVISION`, which should be compatible with most/all external repos. This means the decision to relink on a Git status change will be left up to the root workspace.